app/models/crop_rotation.py

```diff
@@ -0,0 +1,39 @@
+import uuid
+from datetime import date  # noqa
+from typing import TYPE_CHECKING
+
+from sqlalchemy import Column, Date, Float, ForeignKey, String, Text
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import relationship
+
+from app.db.base_class import Base
+
+if TYPE_CHECKING:
+    from app.models import Research
+
+
+class CropRotation(Base):
+    """The Crop Rotation table contains the information about the crop rotation."""
+
+    __tablename__ = "crop_rotation"
+
+    id: uuid.UUID = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    planting_date: date = Column(Date, nullable=False)  # noqa: F811
+    harvest_date: date = Column(Date, nullable=False)  # noqa: F811
+    crop_yield: float = Column(Float, nullable=True)
+    yield_unit: str = Column(String(50), nullable=True)
+    seeding_rate: float = Column(Float, nullable=True)
+    seeding_rate_unit: str = Column(String(50), nullable=True)
+    fertilizer_application: str = Column(Text, nullable=True)
```

Comment:
We could also consider an array to capture the fertilizer events since there may be years where there are multiple applications. Something like:

```
{
[
fertilizer_date: 2015-04-29
fertilizer_rate: 20 lbs / acre
fertilizer_type: UAN (32%) - the starter
],
[
fertilizer_date: 2015-05-29
fertilizer_rate: 150 lbs / acre
fertilizer_type: UAN (32%) - the side dress
]
}
```

Reply:
Ahh yes. I thought of that but he had a description of what and how he applied so I converted it into a text field so I can add that entire explanation. So in the data file, it looks like this.

`"fertilizer_application": "Total of 180 lbs/A (202 kg/ha) split applied as follows: 20 lbs of N/A as starter with planter (combination 2x2 and pop-up) and 160 lbs of N/A as UAN (28%) side-dress on 05-29-2015"`

Reply:
This would capture all that, along with how he has done. Do you still want me to add those other fields?

Reply:
We could add a description field somewhere to capture the explanation too. I think having more metadata is going to be better and this starts to look like a proper farm/field description of events that could later be used to generate an input file for something like DSSAT (crop simulation).

Reply:
Okay, that sounds good then. I will add that.

Reply:
I think it's better to capture it into fields that separate out the information rather than having to parse from a text description. Maybe in the short run it's good enough to pull out what we need, but it's more brittle than having distinct fields.